robot-name/example.js

```diff
@@ -1,47 +1,27 @@
-const HOW_MANY_CHARS = 26;
-const HOW_MANY_NUMBERS = 1000;
-const CHAR_CODE_A = 'A'.charCodeAt(0);
+const alpha = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+  usedNames ={};
 
-let generatedNames = {};
+const random = (max) => Math.floor(Math.random()*max);
 
-function generateName() {
-  let name = '';
-  name += generateRandomChar();
-  name += generateRandomChar();
-  name += pad(generateRandomNumber());
-
-  return getNotGeneratedName(name);
-}
-
-function generateRandomChar() {
-  const random = Math.floor(Math.random() * HOW_MANY_CHARS);
-  return String.fromCharCode(CHAR_CODE_A + random);
-}
-
-function pad(number) {
-  return ('000' + number).slice(-3);
-}
-
-function generateRandomNumber() {
-  return Math.floor(Math.random() * HOW_MANY_NUMBERS);
-}
-
-function getNotGeneratedName(name) {
-  if (generatedNames[name]) {
-    return generateName();
-  }
-
-  generatedNames[name] = true;
+const generateName = () => {
+  let name = alpha.charAt(random(26))
+    +alpha.charAt(random(26))
+    +random(10)+random(10)+random(10);
+  usedNames[name] ? name = generateName() : usedNames[name] = true;
   return name;
-}
+};
 
 export default class Robot {
-  constructor() {
-    this.name = generateName();
+  constructor(){
+    this.robotName = generateName();
+  }
+
+  get name() {
+    return this.robotName;
   }
 
   reset() {
-    this.name = generateName();
+    this.robotName = generateName();
+    return this;
```

Comment:
Is `reset()` required to return `this`? If specs don't use the returned value, why not just simple return nothing?